### ISSUE TITLE
Polygons opacity rendering erratically on Tablets 

### ISSUE THREAD

**Issue by payros** [Leaflet/Leaflet#3959]
I have a map with geoJSON features, and when I click on a feature, it calls setStyle and changes the opacity of that feature. However, on a galaxy tab, when I click on a feature, rectangles appear across parts of my polygons. As seen on the picture below:
![leaflet_rendering_issue](https://cloud.githubusercontent.com/assets/14399593/10678025/bd8d6044-78dd-11e5-9621-3dd532837b93.png)

On this picture I've clicked on the blue polygon near bottom left and the rectangle appeared on the right side of the screen.

Any idea what this could be? It happens both on Chrome and Firefox.

**Comment by IvanSanchez** [comment]
Hum.

- Are you using SVG renderer or Canvas renderer? Have you tried switching the renderer?
- Can you reproduce on any polygon? Some? Are they properly closed? Does it reproduce if you only add those polygons to the map?
- Is there a minimal public webpage we can try to reproduce this?
- Does it reproduce on any other devices?
- If you inspect chrome with a desktop chrome, do you see extra SVG polygons somewhere?

**Comment by payros** [comment]
Are you using SVG renderer or Canvas renderer? Have you tried switching the renderer? **SVG.** **Switching to canvas seemed to have fixed it**
Can you reproduce on any polygon? Some? Are they properly closed? Does it reproduce if you only add those polygons to the map?**I Can reproduce on any polygon. Depending on which polygon I select, the rectangle will change shape or appear in a different location. They are all properly closed.** 
Is there a minimal public webpage we can try to reproduce this? **I'll try to add it somewhere and post a link**
Does it reproduce on any other devices? **I've tested on 4 identical tablets (SM-P605V). I do not have access to other models.**
If you inspect chrome with a desktop chrome, do you see extra SVG polygons somewhere? **I went through every SVG polygon on chrome desktop, and they do not show up there.**

**Comment by IvanSanchez** [comment]
For posting a public example, there's playground-leaflet.rhcloud.com which should be easy to use.

What specific versions of chrome and firefox are you using? Android version? Can you try in an android phone?

**Comment by mourner** [comment]
Also, what Leaflet version is affected? 0.7.5? 1.0 beta-2?

**Comment by payros** [comment]
Versions:
Leaflet - **0.7.5**
Android - **4.4.2**
Chrome - **33.0.1750.517**

I created this simplified version of my map here:
http://playground-leaflet.rhcloud.com/tib/edit?html,js,output

But apparently I simplified too much because I could only reproduce the problem once(After I added the line that calls .bringToFront() ). I'll add more elements from my original map to see if I can get it to happen consistently.

**Comment by payros** [comment]
Okay, I was able to reproduce it a little better on this one:
http://playground-leaflet.rhcloud.com/xac/1/edit?html,js,output

It issue seems to occur only when both width and height of the map container are set using relative units. Also, I noticed it only occurs if you're zoomed in to at least 13.

Finally, it seems that if you pan the map to the left or right, the map gets rendered and the rectangles go away.
